cmd/internal/ecs/ecs.go

```diff
@@ -91,40 +91,29 @@ func init() {
 
 func addAgentToECS(cmd *cobra.Command, args []string) error {
 	// Check for API key
-	_, _, err := cmderr.RequireAkitaAPICredentials("The Postman Live Collections Agent must have an API key in order to capture traces.")
+	_, err := cmderr.RequirePostmanAPICredentials("The Postman Live Collections Agent must have an API key in order to capture traces.")
 	if err != nil {
 		return err
 	}
 
-	// Check project's existence
-	if projectFlag == "" {
-		return errors.New("Must specify the name of your Akita project with the --project flag.")
+	// Check collecton Id's existence
+	if collectionId == "" {
+		return errors.New("Must specify the id of your collection with the --collection flag.")
 	}
 	frontClient := rest.NewFrontClient(rest.Domain, telemetry.GetClientID())
-	_, err = util.GetServiceIDByName(frontClient, projectFlag)
+	_, err = util.GetServiceIDByPostmanCollectionID(frontClient, context.Background(), collectionId)
 	if err != nil {
-		// TODO: we _could_ offer to create it, instead.
-		if strings.Contains(err.Error(), "cannot determine project ID") {
-			return cmderr.AkitaErr{
-				Err: fmt.Errorf(
-					"Could not find the project %q in the Akita cloud. Please create it from the Akita web console before proceeding.",
-					projectFlag,
-				),
-			}
-		} else {
-			return cmderr.AkitaErr{
-				Err: errors.Wrapf(
-					err,
-					"Could not look up the project %q in the Akita cloud",
-					projectFlag,
-				),
-			}
+		return cmderr.AkitaErr{
+			Err: fmt.Errorf(
+				"could not find the serviceId for %q",
```

Comment:
This doesn't mean anything to the user; try to avoid showing "serviceID".  There are three cases to consider:

1. The collection ID doesn't exist.
2. The API key presented doesn't have access (or I suppose the API key is absent?)
3. There was an internal problem creating the new service and they need support.

I don't think we can distinguish between 1 + 2, but we should have a friendly error message specifying that the problem is almost certainly their key.  (And if they didn't specify one at all, make clear that that is the problem!)

Reply:
Yeah I was thinking of discussing the same.

Reply:
The correct error states handling is being already done by the function `GetOrCreateServiceIDByPostmanCollectionID`